lib/mayaUsd/render/vp2RenderDelegate/mesh.cpp

```diff
@@ -1597,7 +1642,8 @@ void HdVP2Mesh::_UpdateDrawItem(
     }
 #endif
 
-    if (desc.geomStyle == HdMeshGeomStyleHull) {
+    if (desc.geomStyle == HdMeshGeomStyleHull
+        && desc.shadingTerminal == HdMeshReprDescTokens->surfaceShader) {
```

Comment:
defaultMaterial repr has a different shading terminal and will ship the shading update.
Regular smoothMesh has "surfaceShader" as terminal, so this addition does not require version check.

Reply:
If we modify the real material while in default material mode does it update correctly when we turn off "use default material"? IIRC somewhere we clear all the dirty flags, so we'd forget about the dirty material id and then not update.